interfaces/builtin/polkit.go

```diff
@@ -60,6 +60,12 @@ dbus (send)
     interface="org.freedesktop.PolicyKit1.Authority"
     member="{,Cancel}CheckAuthorization"
     peer=(name="org.freedesktop.PolicyKit1", label=unconfined),
+dbus (send)
+    bus=system
+    path="/org/freedesktop/PolicyKit1/Authority"
+    interface="org.freedesktop.PolicyKit1.Authority"
+    member="RegisterAuthenticationAgentWithOptions"
```

Comment:
@alexmurray what are the security implications of this for this if any? is this something any polkit client might want to do?

Reply:
this is a good question @pedronis - I suspect this allows an application to register to be the authentication backend for policykit but I couldn't find a good description of this dbus method - but assuming that is about right, this sounds like a very privileged thing. I then went and looked at the fwupd source code and couldn't find anything that would appear to need this - only interaction with polkit itself as a client - so I wonder if this is actually needed? It feels like it shouldn't be here so should likely be removed but it would be good to get clarification from @valentindavid / @superm1

Reply:
I agree it can be removed.  Fwupd doesn't use `polkit_authority_register_authentication_agent*` at all.

Reply:
What about `pkttyagent` in https://github.com/fwupd/fwupd/blob/main/src/fu-polkit-agent.c for the no-GUI usecase?

Reply:
Good point ☝️

Reply:
Ah thanks - so the fwupd snap includes and uses the pkttyagent binary from polkit itself - makes sense then to require this  permission.